python/langsmith/client.py

```diff
@@ -645,16 +592,128 @@ def create_run(
         runtime_env = ls_env.get_runtime_and_metrics()
         run_extra["runtime"] = {**runtime_env, **runtime}
         headers = {**self._headers, "Accept": "application/json"}
+        return {
+            "data": json.dumps(run_create, default=_serialize_json),
+            "headers": headers,
+            "timeout": self.timeout_ms / 1000,
+        }
+
+    def create_run(
+        self,
+        name: str,
+        inputs: Dict[str, Any],
+        run_type: str,
+        *,
+        execution_order: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Persist a run to the LangSmith API.
+
+        Parameters
+        ----------
+        name : str
+            The name of the run.
+        inputs : Dict[str, Any]
+            The input values for the run.
+        run_type : str
+            The type of the run, such as tool, chain, llm, retriever,
+            embedding, prompt, or parser.
+        execution_order : int or None, default=None
+            The position of the run in the full trace's execution sequence.
+                All root run traces have execution_order 1.
+        **kwargs : Any
+            Additional keyword arguments.
+
+        Raises
+        ------
+        LangSmithUserError
+            If the API key is not provided when using the hosted service.
+        """
+        request_kwargs = self._prepare_run_create(
+            name,
+            inputs,
+            run_type,
+            execution_order=execution_order,
+            **kwargs,
+        )
         self.request_with_retries(
             "post",
             f"{self.api_url}/runs",
-            request_kwargs={
-                "data": json.dumps(run_create, default=_serialize_json),
-                "headers": headers,
-                "timeout": self.timeout_ms / 1000,
-            },
+            request_kwargs=request_kwargs,
+        )
+
+    async def acreate_run(
+        self,
+        name: str,
+        inputs: Dict[str, Any],
+        run_type: str,
+        *,
+        execution_order: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Persist a run to the LangSmith API.
+
+        Parameters
+        ----------
+        name : str
+            The name of the run.
+        inputs : Dict[str, Any]
+            The input values for the run.
+        run_type : str
+            The type of the run, such as tool, chain, llm, retriever,
+            embedding, prompt, or parser.
+        execution_order : int or None, default=None
+            The position of the run in the full trace's execution sequence.
+                All root run traces have execution_order 1.
+        **kwargs : Any
+            Additional keyword arguments.
+
+        Raises
+        ------
+        LangSmithUserError
+            If the API key is not provided when using the hosted service.
+        """
+        request_kwargs = self._prepare_run_create(
```

Comment:
Lots of instances like this where prep code is shared and then the request method differs. Could maybe reduce LoC by doing a decorator or something but it's a bit ugly